presto-main/src/main/java/com/facebook/presto/cost/VariableStatsEstimate.java

```diff
@@ -79,6 +84,16 @@ public VariableStatsEstimate(
         checkArgument(distinctValuesCount >= 0 || isNaN(distinctValuesCount), "Distinct values count should be non-negative, got: %s", distinctValuesCount);
         // TODO normalize distinctValuesCount for an empty range (or validate it is already normalized)
         this.distinctValuesCount = distinctValuesCount;
+        this.histogram = requireNonNull(histogram, "histogram is null");
+    }
+
+    public VariableStatsEstimate(double lowValue,
+            double highValue,
+            double nullsFraction,
+            double averageRowSize,
+            double distinctValuesCount)
+    {
+        this(lowValue, highValue, nullsFraction, averageRowSize, distinctValuesCount, new UniformDistributionHistogram(lowValue, highValue));
```

Comment:
Instead of assign a UniformDistributionHistogram, can you make the histogram to be an optional field and assign empty() when histogram is turned off?

Reply:
fixed